sys/net/include/etx_beaconing.h

```diff
@@ -78,41 +92,66 @@ extern "C" {
  *                  denoting the amount of packets received from that IP address
  *
  * We only need 1 octet for the ip address since RPL for now only allows for
- * 255 different addresses.
+ * 255 different addresses
  *
  * If the length of this packet says 0, it has received no other beaconing
- * packets itself so far.
+ * packets itself so far
  *
  * The packet is always 32bytes long, but may contain varying amounts of
- * information.
- * The information processed shall not exceed the value set in Option Length.
+ * information
+ * The information processed shall not exceed the value set in Option Length
  */
-typedef struct __attribute__((packed)) etx_probe_t {
-    uint8_t code;
-    uint8_t length;
-    uint8_t data[30];
+ typedef struct __attribute__((packed)) etx_probe_t {
+    uint8_t code;                       /**< Option type */
+    uint8_t length;                     /**< Option Length*/
+    uint8_t data[30];                   /**< Option Data */
 } etx_probe_t;
-
+/**
+ * @brief etx neighbour data structure
+ */
 typedef struct etx_neighbor_t {
-    ipv6_addr_t addr;           //The address of this node
-    uint8_t     tx_cur_round;   //The indicator for receiving a packet from this candidate this round
-    uint8_t     packets_tx[ETX_WINDOW]; //The packets this node has transmitted TO ME
-    uint8_t     packets_rx;     //The packets this node has received FROM ME
-    double      cur_etx;        //The currently calculated ETX-value
-    uint8_t     used;           //The indicator if this node is active or not
+    /** The address of this node */
+    ipv6_addr_t addr;
+    /** The indicator for receiving a packet from this candidate this round */
+    uint8_t     tx_cur_round;
+    /** The packets this node has transmitted TO ME */
+    uint8_t     packets_tx[ETX_WINDOW];
+    /** The packets this node has received FROM ME */
+    uint8_t     packets_rx;
+    /** The currently calculated ETX-value */
+    double      cur_etx;
+     /** The indicator if this node is active or not */
+    uint8_t     used;
 } etx_neighbor_t;
 
-//prototypes
+ /**
+  * @brief initializes the module
+  * @param[in] address a ipv6_addr_t pointer
+  */
 void etx_init_beaconing(ipv6_addr_t *address);
+
+ /**
+  * @brief calculates the ETX-value
+  * @param[in] address a ipv6_addr_t pointer
+  * @return the calculated ETX-value
+  */
 double etx_get_metric(ipv6_addr_t *address);
+
+ /**
+  * @brief Updates the current ETX value of a candidate
+  * @param[in] neighbour an etx_neighbour_t pointer
```

Comment:
the parameter is actually an inout since it is both being used for input (candidate->packets_rx, candidate->packets_tx indirectly inside etx_count_packet_tx) as well as being written to (candidate->cur_etx).